workflow/controller/workflowpod.go

```diff
@@ -138,6 +157,10 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 		pod.ObjectMeta.Labels[common.LabelKeyControllerInstanceID] = woc.controller.Config.InstanceID
 	}
 
+	if common.GoogleSecretName != "" {
```

Comment:
> secret selector in the configmap instead of environment variable.

Actually, instead of secret selector, I think it would be an LocalObjectReference (https://github.com/kubernetes/kubernetes/blob/d20912c83fe3b9536098920ae279d2201a424c43/pkg/apis/core/types.go#L3977). So the configmap would look like:
```
    artifactRepository:
      gcs:
        credentialsSecret:
          name: my-google-secret
```

And my-google-secret would have:

```
metadata:
  name: my-google-secret
data:
  key.json: <secret value>
```